services/management/ia/alerts_service.go

```diff
@@ -116,12 +116,19 @@ func (s *AlertsService) ListAlerts(ctx context.Context, req *iav1beta1.ListAlert
 				return err
 			})
 			if e != nil {
+				// Only FindRuleByID from transaction above can return codes.NotFound
+				if st, ok := status.FromError(e); ok && st.Code() == codes.NotFound {
```

Comment:
When a rule deleted associated alert may be still present for several seconds (alertmanager delay). So instead of returning error better just to skip such alerts.